src/main/resources/graphql/service/quiz.graphqls

```diff
@@ -134,10 +134,226 @@ type MultipleChoiceAnswer {
+    """
+    showBlanksList: Boolean!
+
+    # inherited from Question
+    """
+    Unique identifier of the question.
+    """
+    id: UUID!
+    """
+    Number of the question, i.e., the position of the question in the list of questions.
+    Only relevant if questionPoolingMode is ORDERED.
+    """
+    number: Int!
+    """
+    Type of the question.
+    """
+    type: QuestionType!
+    """
+    Optional hint for the question, can be markdown.
+    """
+    hint: String
+}
+
+
+interface ClozeElement {
+    """
+    Type of the element.
+    """
+    type: ClozeElementType!
+}
+
+type ClozeTextElement implements ClozeElement {
+    """
+    Text of the element, can be markdown.
+    """
+    text: String!
+
+    # inherited from ClozeElement
+    """
+    Type of the element. Must be TEXT.
+    """
+    type: ClozeElementType!
+}
+
+type ClozeBlankElement implements ClozeElement {
+    """
+    The correct answer for the blank.
+    """
+    correctAnswer: String!
+    """
+    Wrong answers for the blank.
+    """
+    wrongAnswers: [String!]!
+    """
+    Feedback for the blank when the user selects a wrong answer, can be markdown.
+    """
+    feedback: String
+
+    # inherited from ClozeElement
+    """
+    Type of the element. Must be BLANK.
+    """
+    type: ClozeElementType!
+}
+
+enum ClozeElementType {
+    TEXT
+    BLANK
+}
+
+"""
+Association question, i.e., a question where the user has to assign the correct right side to each left side.
+"""
+type AssociationQuestion implements Question {
+    """
+    Text to display above the assignment question, can be markdown.
+    """
+    text: String!
+    """
+    List of correct associations.
+    """
+    correctAssociations: [SingleAssociation!]!
+    """
+    Computed list of all the left sides of the associations, shuffled.
+    """
+    leftSide: [String!]!
+    """
+    Computed list of all the right sides of the associations, shuffled.
+    """
+    rightSide: [String!]!
+
+    # inherited from Question
+    """
+    Unique identifier of the question.
+    """
+    id: UUID!
+    """
+    Number of the question, i.e., the position of the question in the list of questions.
+    Only relevant if questionPoolingMode is ORDERED.
+    """
+    number: Int!
+    """
+    Type of the question.
+    """
+    type: QuestionType!
+    """
+    Optional hint for the question, can be markdown.
+    """
+    hint: String
+}
+
+type SingleAssociation {
+    """
+    The left side of the assignment.
+    """
+    left: String!
+    """
+    The right side of the assignment.
+    """
+    right: String!
+
+    """
+    Feedback for the assignment when the user assigns a wrong answer, can be markdown.
+    """
+    feedback: String
+}
+
+"""
+A question with a clear, correct answer that can be automatically checked.
+Differs from self-assessment questions in that the user has to enter one of the correct answers and
+the answer is checked automatically.
+"""
+type FreeTextQuestion {
```

Comment:
e.g. math questions or other clearly defined answers

Reply:
i find the name a bit confusing because Freitext usually implies (IMO) that there isn't an exact answer that can be checked 1:1

Reply:
true, do you have a better idea?

Reply:
Maybe just TextInputQuestion?